scripts/pull_translations.rb

```diff
@@ -0,0 +1,167 @@
+      puts "Removed translation %s" % translation if show_info
+    end
+  end
+  File.write(filename, strings_doc.to_xml)
+end
+
+Struct.new(
+  'Translation',
+  :translated_string,
+  :description_comment,
+  :string_node
+)
+
+def make_translations_hash(filename, included_strings)
+  strings_doc = File.open(filename) { |f| Nokogiri::XML(f) }
+  translation_nodes = strings_doc.xpath('//string').to_a
+  translations_hash = {}
+  translation_nodes.each do |translation|
+    translation_comment = nil
+    if translation.previous.previous.comment?
+      translation_comment = translation.previous.previous
+    end
+    if included_strings.include? translation['name']
+      translations_hash[translation['name']] = Struct::Translation.new(
+        translation.content, translation_comment, translation
+      )
+    end
+  end
+  translations_hash
+end
+
+def replace_updated_translations(filename, translations_hash, show_info)
+  puts '▸ Updating strings that have changed'
+  strings_doc = File.open(filename) { |f| Nokogiri::XML(f) }
+  translation_nodes = strings_doc.xpath('//string').to_a
+  translation_nodes.each do |translation|
+    next unless translations_hash.key?(translation['name'])
+    key = translation['name']
+    next unless !translation.previous.previous.nil? &&
+                translation.previous.previous.comment?
+    translation_comment = translation.previous.previous
+    unless translation_comment.text.eql? translations_hash[key].description_comment.text
+      puts "Updating translation comment for %s" % translation_comment if show_info
+      translation_comment.replace(translations_hash[key].description_comment)
+    end
+    unless translation.content.eql? translations_hash[key].translated_string
+      puts "Updating translation to %s" % translation if show_info
+      translation.content = translations_hash[key].translated_string
+    end
+  end
+  File.write(filename, strings_doc.to_xml)
+end
+
+def add_new_translations(filename, translations_hash, show_info)
+  puts '▸ Adding new translations'
+  strings_doc = File.open(filename) { |f| Nokogiri::XML(f) }
+  translation_nodes = strings_doc.xpath('//string')
+  existing_strings = strings_doc.xpath('//string/@name').to_a.to_set(&:text)
+  new_strings = translations_hash.keys.to_set - existing_strings
+  # Grab a new line node to use for formatting
+  new_line_node = translation_nodes.first.next.dup
+  # Move the last new line to the end
+  last_newline_node = translation_nodes.last.next.unlink
+  new_strings.each do |string|
+    translation = translations_hash[string]
+    unless translation.description_comment.nil?
+      strings_doc.root.add_child(new_line_node.dup)
+      strings_doc.root.add_child(translation.description_comment)
+    end
+    strings_doc.root.add_child(new_line_node.dup)
+    strings_doc.root.add_child(translation.string_node)
+    puts "Added translation for %s" % translation.string_node if show_info
```

Comment:
Consider using a string template like `"Added translation for #{translation.string_node}"`